### PR TITLE
Fix the build

### PR DESCRIPTION
**Pull request by willprice**
A variety of fixes to make circle CI green again

- Bump 16.04 -> 20.04 (16.04 is now out of support)
- Bump 20.04 -> 22.04
- Ditch usage of &: in makefiles, this is unsupported in GNU make <4.3 and fails silently in older versions (failing back to non grouped target definitions -- very much not the desired behaviour)
- Bump some software versions that had bit rotted

### CODE REVIEW

**Comment by willprice** [comment]
I pulled in the changes from https://github.com/GetPoplog/Seed/pull/122 into this PR.

File: .circleci/config.yml [hunk 1]
```diff
@@ -169,68 +169,68 @@ jobs:
       - run:
           name: 'Install Poplog from *.deb file'
           command: |
-            sudo apt install -y ./_build/artifacts/poplog_*_amd64.deb
+            DEBIAN_FRONTEND=noninteractive sudo apt-get install -y ./_build/artifacts/poplog_*_amd64.deb
```
Comment (willprice) [inline .circleci/config.yml:172]: This was necessary as apt would prompt for user input and hang the build

File: .circleci/config.yml [hunk 2]
```diff
@@ -169,68 +169,68 @@ jobs:
...
-                --volume $PWD:/mnt \
-                centos:8 \
-                /mnt/.circleci/scripts/make_rpm.sh /mnt
-      - store_artifacts:
-          path: _build/artifacts
-      - persist_to_workspace:
-          root: _build
-          paths:
-            - artifacts/*.rpm
+#  build_rpm:
```
Comment (willprice) [inline .circleci/config.yml:185]: Let's address RPM in a follow up PR

**Review by sfkleach — APPROVED** [review]

**Comment by sfkleach** [comment]
Looks solid. Will withdraw earlier PR as this supersedes.

**Pull request merged**